Exercise_8/index.js

```diff
@@ -0,0 +1,24 @@
+import contentService from './src/services/contentService';
+import historyService from './src/services/historyService';
+
+/* global window:true */
+window.onload = () => {
+  contentService.setHomeContent();
+};
+
+window.document.querySelectorAll('.sign-in').forEach((button) => {
+  button.addEventListener('click', () => {
+    historyService.addToHistory('/sign-in');
+    contentService.setSignInContent();
+  }, false);
+});
+
+window.onpopstate = () => {
+  const singInUrl = '/sign-in';
+  const currentUrl = window.location.pathname;
+  if (currentUrl !== singInUrl) {
```

Comment:
According to the "code complete" book, it is better not to have negative conditions if you are able to make it positive.
```
if (currentUrl === singInUrl) {
  contentService.setSignInContent();
} else {
  contentService.setHomeContent();
}
```
And I can tell you why. Imagine you had a separate function to check the condition. And for your case it would be this: `if (!isSingInUrl()) {`. It would start with `!` which makes no sense there. `if (isSingInUrl()) {` is better because you would have the pure condition and you wouldn't have to check the condition and then revert it.